engine/src/main/java/org/hibernate/search/engine/search/timeout/spi/TimeoutManager.java

```diff
@@ -138,12 +141,18 @@ public boolean hasHardTimeout() {
 		return this.type == Type.EXCEPTION;
 	}
 
+	/**
+	 * Returns the time passed from the start with high precision.
+	 * This method may be performance expensive.
+	 *
+	 * @return high precision duration of took time.
+	 */
 	public Duration tookTime() {
-		return Duration.ofMillis( elapsedTimeInMilliseconds() );
+		return Duration.ofMillis( timingSource.nanoTime() - nanoTimeStart );
 	}
 
 	protected long elapsedTimeInMilliseconds() {
```

Comment:
Maybe the reason for having two methods with a similar purpose would be more obvious if you renamed this to `elapsedTimeEstimateMillis()`?

Reply:
Good idea!